copy.go

```diff
@@ -15,17 +15,21 @@ type timespec struct {
 }
 
 // Copy copies src to dest, doesn't matter if src is a directory or a file.
-func Copy(src, dest string, opt ...Options) error {
+func Copy(src, dest string, opts ...Options) error {
+	opt := assureOptions(src, dest, opts...)
 	info, err := os.Lstat(src)
 	if err != nil {
-		return err
+		return onError(err, opt)
 	}
-	return switchboard(src, dest, info, assureOptions(src, dest, opt...))
+	return switchboard(src, dest, info, opt)
 }
 
 // switchboard switches proper copy functions regarding file type, etc...
 // If there would be anything else here, add a case to this switchboard.
 func switchboard(src, dest string, info os.FileInfo, opt Options) (err error) {
+	defer func() {
+		err = onError(err, opt)
+	}()
```

Comment:
I don't think we should use `defer` here. There are only 2 points of return, so let's keep it simple.